src/main/java/org/skriptlang/skript/lang/converter/Converters.java

```diff
@@ -349,13 +353,35 @@ public static <From, To> To convert(@Nullable From from, Class<? extends To>[] t
 				return (To) from;
 		}
 
+		// Look through all converters
 		Class<From> fromType = (Class<From>) from.getClass();
 		for (Class<? extends To> toType : toTypes) {
 			Converter<From, ? extends To> converter = getConverter(fromType, toType);
 			if (converter != null)
 				return converter.convert(from);
 		}
 
+		/**
+		 * Attempts to convert most likely a literal.
+		 * Example of this can be setting a variable to a snowball. Skript assumes it's an itemtype.
+		 * Now when using that variable in an entitydata expression, Skript attempts to convert but can't
+		 * as no converters exist for ItemType -> EntityData. That's where this chunk of code steps in.
+		 * 
+		 * This section of code has proven to convert that said issue to the EntityData and allow for proper usage.
+		 * This will only happen when the names of the literals are the same of two classinfos and mainly for Variables.
+		 * 
+		 * See https://github.com/SkriptLang/Skript/issues/5493
+		 */
+		for (Class<? extends To> toType : toTypes) {
```

Comment:
Are there other places where this might create unwanted type conformity? (E.g. when we expect something to error because it's wrong but it would now succeed.)

Reply:
It turns out there are - it creates cases where syntax will work for the wrong kind of thing (e.g. strings and entities as items)

Reply:
Moved to variables to avoid incorrect casting of everything